clang/test/C/C23/n3042.c

```diff
@@ -93,6 +91,9 @@ void test() {
   (int *)null_val;    // ok
   (int *)nullptr;     // ok
   (nullptr_t)nullptr; // ok
+  (nullptr_t)0;       // ok
+  (nullptr_t)(void *)0; // ok
+  (nullptr_t)null_val;  // ok
```

Comment:
What about `(nullptr_t)__null`

Reply:
I've added a test for that.

Reply:
Oops, I take that back, C doesn't have `__null`, that's a C++ism: https://godbolt.org/z/1sKsEM6dY

> `KEYWORD(__null                      , KEYCXX)`

Reply:
Oh interesting; my bad, I thought it was an extension for both C and C++.

Reply:
I thought that as well!